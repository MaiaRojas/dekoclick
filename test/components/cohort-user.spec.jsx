import React from 'react';
import { shallow, mount } from 'enzyme';
import CohortUser from '../../src/components/cohort-user';


describe('<CohortUser />', () => {

  it('should warn when missing required props', () => {
    const spy = jest.spyOn(console, 'error').mockImplementation(jest.fn());
    shallow(<CohortUser />);
    expect(spy.mock.calls).toMatchSnapshot();
    spy.mockReset();
    spy.mockRestore();
  });

  it.skip('should ...', () => {
    /* eslint-disable jsx-a11y/aria-role */
    const component = mount((
      <CohortUser
        uid="xxx"
        cohortid="lim-2050-01-bc-core-am"
        role="student"
        profile={{
          name: 'Ada Lovelace',
          email: 'ada@lovelace.com',
          github: 'lupomontero',
        }}
        toggleMoveDialog={jest.fn()}
        firebase={{ database: jest.fn() }}
      />
    ));
    /* eslint-enable jsx-a11y/aria-role */

    console.log(component.find('a').map(el => console.log(el) || el.text()));
    // console.log(component.html());
    // expect(component.html()).toMatchSnapshot();
  });

});
