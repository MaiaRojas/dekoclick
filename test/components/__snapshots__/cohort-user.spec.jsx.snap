// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CohortUser /> should warn when missing required props 1`] = `
Array [
  Array [
    "Warning: Failed prop type: The prop \`uid\` is marked as required in \`CohortUser\`, but its value is \`undefined\`.
    in CohortUser",
  ],
  Array [
    "Warning: Failed prop type: The prop \`cohortid\` is marked as required in \`CohortUser\`, but its value is \`undefined\`.
    in CohortUser",
  ],
  Array [
    "Warning: Failed prop type: The prop \`cohortUser\` is marked as required in \`CohortUser\`, but its value is \`undefined\`.
    in CohortUser",
  ],
  Array [
    "Warning: Failed prop type: The prop \`profile\` is marked as required in \`CohortUser\`, but its value is \`undefined\`.
    in CohortUser",
  ],
  Array [
    "Warning: Failed prop type: The prop \`toggleMoveDialog\` is marked as required in \`CohortUser\`, but its value is \`undefined\`.
    in CohortUser",
  ],
  Array [
    "Warning: Failed prop type: The prop \`firebase\` is marked as required in \`CohortUser\`, but its value is \`undefined\`.
    in CohortUser",
  ],
  Array [
    "Warning: Failed prop type: The prop \`parsedCohortId\` is marked as required in \`CohortUser\`, but its value is \`undefined\`.
    in CohortUser",
  ],
]
`;
